docs(sdk/java): api

eunomie opened the pull request:
Documentation about arguments, custom functions, return values and others for the Java SDK

vikram-dagger commented on docs/current_docs/api/custom-functions.mdx:
````diff
@@ -63,6 +63,16 @@ Update the `src/MyModule.php` file with the following code:
+
+Update the `src/main/java/io/dagger/modules/mymodule/MyModule.java` file with the following code:
+
+```java file=./snippets/functions/functions-complex/java/src/MyModule.php
````
CI is failing as this path is incorrect: `ENOENT: no such file or directory, open '/opt/build/repo/docs/current_docs/api/snippets/functions/functions-complex/java/src/MyModule.php'`

eunomie replied:
Thanks, it should be fixed now

vikram-dagger reviewed (changes requested):
See CI failure notice

vikram-dagger commented on docs/current_docs/api/arguments.mdx:
```diff
@@ -176,6 +188,15 @@ Here is an example of a Dagger function that accepts an integer argument:
 
+<TabItem value="Java">
+:::note
+You can either use the primitive `int` type or the boxed `java.lang.Integer` type.
```
Error: https://dagger.cloud/dagger/traces/15eaad8737d00a3bda6529c50445287b

vikram-dagger commented on docs/current_docs/api/arguments.mdx:
````diff
@@ -226,6 +247,15 @@ The imported `float` type is a `number` underneath, so you can use it as you wou
+You can either use the primitive `float` type or the boxed `java.lang.Float` type.
+:::
+
+```java file=./snippets/functions/arguments-return-values-float/java/src/main/java/io/dagger/modules/mymodule/MyModule.java
````
Error https://dagger.cloud/dagger/traces/15eaad8737d00a3bda6529c50445287b

eunomie replied:
Fixed in #9647

vikram-dagger commented on docs/current_docs/api/arguments.mdx:
````diff
@@ -267,6 +297,15 @@ The PHP SDK needs the typing information at runtime to correctly report to the A
+For instance instead of the `String[] names` argument in the example, you can have `List<String> names`.
+:::
+
+```java file=./snippets/functions/arguments-array/java/src/main/java/io/dagger/modules/mymodule/MyModule.java
````
Error https://dagger.cloud/dagger/traces/6131e753f919fd3d0e183d804e7a85a1

eunomie replied:
this works for me with `main` version

vikram-dagger commented on docs/current_docs/api/arguments.mdx:
````diff
@@ -558,6 +615,16 @@ To specify a function argument as optional, simply make it nullable. When using
+When an argument is not set as optional, Dagger will ensure the value is not `null` by adding a call to `Objects.requireNonNull` against the argument.
+:::
+
+```java file=./snippets/functions/arguments-optional/java/src/main/java/io/dagger/modules/mymodule/MyModule.java
````
Also error https://dagger.cloud/dagger/traces/04c0af439d32debcbba6da2263b2ab16

eunomie replied:
Optional support has been merged in #9605 
Working for me with `main` version

vikram-dagger commented on docs/current_docs/api/custom-functions.mdx:
````diff
@@ -63,6 +64,16 @@ Update the `src/MyModule.php` file with the following code:
+```java file=./snippets/functions/functions-complex/java/src/main/java/io/dagger/modules/mymodule/MyModule.java
+```
+
+Dagger Functions must be public. The function needs to be decorated with the `@Function` annotation
````
```suggestion
Dagger Functions must be public. The function must be decorated with the `@Function` annotation
```

vikram-dagger commented on docs/current_docs/partials/_dagger_module_init.mdx:
```diff
@@ -114,6 +114,45 @@ In this structure:
+    - `entrypoint` contains the generated entrypoint for the module.
+
+The `target` folder is re-generated every time you run `dagger develop` and allows to have access to code completion
+and types from the IDE.
```
```suggestion
and type hinting in the IDE.
```

vikram-dagger commented:
It would be cleaner to merge this into `docs-java`

![image](https://github.com/user-attachments/assets/f349e805-cdac-4e15-a955-636782bc6cb0)

eunomie commented on docs/current_docs/api/return-values.mdx:
````diff
@@ -122,6 +134,15 @@ The imported `float` type is a `number` underneath, so you can return it as you
+You can either use the primitive `float` type or the boxed `java.lang.Float` type.
+:::
+
+```java file=./snippets/functions/arguments-return-values-float/java/src/main/java/io/dagger/modules/mymodule/MyModule.java
````
same as from `arguments.mds`

eunomie commented:
@vikram-dagger I've updated the base branch.

I don't have access the the traces: `input: trace unauthorized`. If you can copy the errors that would be easier.
Also notice that the documentation was written based on features that has been merged only a few minutes ago, so if you tried from `main` the failure could have been expected.

I've tested the different ones for which you got failures and they are all working on my side.

I also added the different fix/suggestions.

vikram-dagger reviewed (approved):
Approving this but not merging atm as I'm not sure if we should wait until there is a release version with the last fixes.

jpadams commented:
> Approving this but not merging atm as I'm not sure if we should wait until there is a release version with the last fixes.

Looks like release planned for Thursday this week

eunomie commented:
Is the PR good to go? I've seen the quickstart for instance is now in Java, but not the details about the API (this PR)
